homeassistant/components/netgear_lte/__init__.py

```diff
@@ -137,62 +148,116 @@ def get_modem_data(self, config):
+            breaks_in_ha_version="2023.9.0",
+            is_fixable=False,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
         )
 
-        # Binary Sensor
-        binary_sensor_conf = lte_conf[CONF_BINARY_SENSOR]
-        discovery_info = {
-            CONF_HOST: lte_conf[CONF_HOST],
-            CONF_BINARY_SENSOR: binary_sensor_conf,
-        }
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Netgear LTE from a config entry."""
+    host = entry.data[CONF_HOST]
+    password = entry.data[CONF_PASSWORD]
+
+    websession = async_create_clientsession(hass, cookie_jar=CookieJar(unsafe=True))
+
+    hass.data[DOMAIN] = LTEData(websession)
+
+    modem = Modem(
+        hostname=host,
+        password=password,
+        websession=websession,
+    )
+    try:
+        await modem.login()
+    except Error as ex:
+        raise ConfigEntryNotReady("Cannot connect/authenticate") from ex
+
+    # Set up modem
+    await _setup_lte(hass, entry.data)
+
+    await async_setup_services(hass)
+
+    hass.async_create_task(
+        discovery.async_load_platform(
+            hass,
+            Platform.NOTIFY,
+            DOMAIN,
+            {CONF_HOST: entry.data[CONF_HOST], CONF_NAME: entry.title},
+            hass.data[DATA_HASS_CONFIG],
+        )
+    )
+    if lte_configs := hass.data[DATA_HASS_CONFIG].get(DOMAIN, []):
+        async_create_issue(
+            hass,
+            DOMAIN,
+            "deprecated_notify",
+            breaks_in_ha_version="2023.9.0",
+            is_fixable=False,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_notify",
+            translation_placeholders={
+                "name": f"{Platform.NOTIFY}.{entry.title.lower().replace(' ', '_')}"
+            },
+        )
         hass.async_create_task(
             discovery.async_load_platform(
-                hass, Platform.BINARY_SENSOR, DOMAIN, discovery_info, config
+                hass,
+                Platform.NOTIFY,
+                DOMAIN,
+                {CONF_HOST: entry.data[CONF_HOST], CONF_NAME: DOMAIN},
+                hass.data[DATA_HASS_CONFIG],
             )
         )
+    for lte_config in lte_configs:
+        if lte_config[CONF_HOST] == entry.data[CONF_HOST]:
+            for notify_conf in lte_config[CONF_NOTIFY]:
+                discovery_info = {
+                    CONF_HOST: lte_config[CONF_HOST],
+                    CONF_NAME: notify_conf.get(CONF_NAME),
+                    CONF_NOTIFY: notify_conf,
+                }
+                hass.async_create_task(
+                    discovery.async_load_platform(
+                        hass,
+                        Platform.NOTIFY,
+                        DOMAIN,
+                        discovery_info,
+                        hass.data[DATA_HASS_CONFIG],
+                    )
+                )
```

Comment:
To make the code clearer, I'd suggest to move the legacy yaml stuff to a helper function, or at least comment the code.